typescript/src/deposit-sweep.ts

```diff
@@ -130,238 +150,297 @@ export async function assembleDepositSweepTransaction(
- * @param walletKeyRing - Key ring created using the wallet's private key.
- * @returns Empty promise.
+ * Signs a P2WSH deposit transaction input and sets the witness script.
+ * @param transaction - The transaction containing the input to be signed.
+ * @param inputIndex - Index pointing to the input within the transaction.
+ * @param deposit - Details of the deposit transaction.
+ * @param prevOutValue - The value from the previous transaction output.
+ * @param keyPair - A Signer object with the public and private key pair.
+ * @returns An empty promise upon successful signing.
  */
 async function signP2WSHDepositInput(
-  transaction: any,
+  transaction: Transaction,
   inputIndex: number,
   deposit: Deposit,
-  walletKeyRing: any
-): Promise<void> {
+  prevOutValue: number,
+  keyPair: Signer
+) {
   const { walletPublicKey, depositScript, previousOutputValue } =
-    await prepareInputSignData(transaction, inputIndex, deposit, walletKeyRing)
+    await prepareInputSignData(deposit, prevOutValue, keyPair)
 
-  const signature: Buffer = transaction.signature(
+  const sigHashType = Transaction.SIGHASH_ALL
+
+  const sigHash = transaction.hashForWitnessV0(
     inputIndex,
     depositScript,
     previousOutputValue,
-    walletKeyRing.privateKey,
-    bcoin.Script.hashType.ALL,
-    1 // segwit sighash version
+    sigHashType
   )
 
-  const witness = new bcoin.Witness()
-  witness.clear()
-  witness.pushData(signature)
-  witness.pushData(Buffer.from(walletPublicKey, "hex"))
-  witness.pushData(depositScript.toRaw())
-  witness.compile()
+  const signature = script.signature.encode(keyPair.sign(sigHash), sigHashType)
+
+  const witness: Buffer[] = []
+  witness.push(signature)
+  witness.push(Buffer.from(walletPublicKey, "hex"))
+  witness.push(depositScript)
 
-  transaction.inputs[inputIndex].witness = witness
+  transaction.ins[inputIndex].witness = witness
 }
 
 /**
- * Creates data needed to sign a deposit input.
- * @param transaction - Mutable transaction containing the input.
- * @param inputIndex - Index that points to the input.
- * @param deposit - Data of the deposit.
- * @param walletKeyRing - Key ring created using the wallet's private key.
- * @returns Data needed to sign the input.
+ * Prepares data for signing a deposit transaction input.
+ * @param deposit - The deposit details.
+ * @param prevOutValue - The value from the previous transaction output.
+ * @param ecPair - A Signer object with the public and private key pair.
+ * @returns A Promise resolving to:
+ * - walletPublicKey: Hexstring representation of the wallet's public key.
+ * - depositScript: Buffer containing the assembled deposit script.
+ * - previousOutputValue: Numeric value of the prior transaction output.
+ * @throws Error if there are discrepancies in values or key formats.
  */
 async function prepareInputSignData(
-  transaction: any,
-  inputIndex: number,
   deposit: Deposit,
-  walletKeyRing: any
+  prevOutValue: number,
+  ecPair: Signer
```

Comment:
`walletKeyPair`?

Reply:
Done in 98b622a5628e24d69b073ce22d713a8995ce0b2a.